src/main/java/hudson/tasks/junit/StdioRetention.java

```diff
@@ -0,0 +1,43 @@
+package hudson.tasks.junit;
+
+import java.util.Locale;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public enum StdioRetention {
+
+    ALL(Messages.StdioRetention_All_DisplayName()),
+    FAILED(Messages.StdioRetention_Failed_DisplayName()),
+    NONE(Messages.StdioRetention_None_DisplayName());
+
+    private final String displayName;
+
+    StdioRetention(String displayName) {
+        this.displayName = displayName;
+    }
+
+    public String getDisplayName() {
+        return displayName;
+    }
+
+    public static final StdioRetention DEFAULT = NONE;
+
+    public static StdioRetention fromKeepLongStdio(boolean value) {
+        return value ? ALL : NONE;
+    }
+
+    public static StdioRetention parse(String value) {
+        if (value == null)
+            return DEFAULT;
```

Comment:
braces around if statements
```suggestion
        if (value == null) {
            return DEFAULT;
        }
```